src/units/attack_type.cpp

```diff
@@ -97,11 +97,15 @@ std::string attack_type::accuracy_parry_description() const
 	return s.str();
 }
 
+static bool check_string(const std::vector<std::string>& filter_attribute, const std::string& attribute)
+{
+	return (std::find(filter_attribute.begin(), filter_attribute.end(), attribute) != filter_attribute.end());
+}
```

Comment:
Use `utils::split_set` for this, and then use the set's `count()` method instead. The same can also be done to simplify `filter_name`'s check.

Reply:
its'done and for filter_range also.